devito/dse/rewriters.py

```diff
@@ -406,6 +411,32 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        Skew the accesses along the time Dimension.
+        """
+
+        skew_factor = self.DEFAULT_SKEW_FACTOR
+
+        _, sub_iterators, directions = cluster.ispace.args
+
+        skew_dim, mapper, intervals = None, {}, []
+
+        for i in cluster.ispace.intervals:
```

Comment:
what would happen here if the time dimension appeared **after** even a single non-time dimension?  you would multiply (line 431) skew_factor times None, which is clearly buggy

I know that basically always you see time, x, y ... in this order, but this is actually not strictly necessary the case

Reply:
Yes, I know what you mean, I should add a typical test with just `Functions`, right?

Reply:
no, it's not (just) about adding tests, I'm saying that this is broken if someone writes (for whatever reason -- and there are reasons) this piece of code:

```
grid = Grid(...)
x, y, z = grid.dimensions
t = grid.stepping_dim
n = Dimension(name='n')
f = Function(name='f', dimensions=(n, t, x, y, z), shape=(...))
...
```

Reply:
Can you please check again, now that I pushed some changes that were not here due to my HOME-pc rebase?
I think you may have missed this change.

Reply:
Also may you provide me with a typical operator for such a problem?

Reply:
following up from the code snippet above

```
...
op = Operator(Eq(f[n, t+1, x, y, z], f[n, t, x, y, z] + 1))
...
```

some "true" tests in `test_subdomains.py`

there are others but I can't now easily pinpoint one